subiquitycore/ui/form.py

```diff
@@ -160,6 +160,19 @@ def __init__(self, field, form, widget):
         if isinstance(widget, WantsToKnowFormField):
             widget.set_bound_form_field(self)
 
+    def is_in_error(self, recurse: bool = True) -> bool:
+        """ Tells whether this field is in error. If recurse is True, we will
+        also check if any of the enabled subform reports an error.
+        """
+        if self.in_error:
+            return True
+
+        if not recurse or not self._enabled \
+                or not isinstance(self.widget, SubFormWidget):
```

Comment:
This check feels wrong. It feels like maybe it would be better to somehow allow the FormField to define what "in error" means for a given situation. This could be a method on FormField that takes a BoundFormField, or it could be overriding bind() in SubFormField to return a subclass of BoundFormField that overrides is_in_error().